Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,216 @@
+        else
+        {
+            QString error = tr("Failed to get engine info.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        QTextStream logStream(&logFile);
+
+        UpdateProgress(tr("Setting Up Environment"));
+
+        auto currentEnvironmentRequest = ProjectUtils::SetupCommandLineProcessEnvironment();
+        if (!currentEnvironmentRequest.IsSuccess())
+        {
+            QStringToAZTracePrint(currentEnvironmentRequest.GetError());
+            return currentEnvironmentRequest;
+        }
+
+        
+
+        m_exportProjectProcess = new QProcess(this);
+        m_exportProjectProcess->setProcessChannelMode(QProcess::MergedChannels);
+        m_exportProjectProcess->setWorkingDirectory(m_projectInfo.m_path);
+
+        
+        QStringList exportArgs = {"export-project", "--export-script", m_projectInfo.m_currentExportScript,
+                                                    "--project-path", m_projectInfo.m_path};
+        QString commandProgram = QDir(engineInfo.m_path).filePath(GetO3DECLIString());
+
+        m_exportProjectProcess->start(commandProgram, exportArgs);
+        if (!m_exportProjectProcess->waitForStarted())
+        {
+            QString error = tr("Exporting project failed to start.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        while (m_exportProjectProcess->waitForReadyRead(MaxExportTimeMSecs))
+        {
+            QString exportOutput = m_exportProjectProcess->readAllStandardOutput();
+
+            logStream << exportOutput;
+            logStream.flush();
+
+            // Show last line of output
+            if (QStringList strs = exportOutput.split('\n', Qt::SkipEmptyParts);
+                !strs.empty())
+            {
+                UpdateProgress(strs.last());
+            }
+
+            if (QThread::currentThread()->isInterruptionRequested())
+            {
+                // QProcess is unable to kill its child processes so we need to ask the operating system to do that for us
+                auto killProcessArgumentsResult = ConstructKillProcessCommandArguments(QString::number(m_exportProjectProcess->processId()));
+                if (!killProcessArgumentsResult.IsSuccess())
+                {
+                    return AZ::Failure(killProcessArgumentsResult.GetError());
+                }
+                auto killProcessArguments = killProcessArgumentsResult.GetValue();
+
+                QProcess killExportProcess;
+                killExportProcess.setProcessChannelMode(QProcess::MergedChannels);
+                killExportProcess.start(killProcessArguments.front(), killProcessArguments.mid(1));
+                bool finishedKilling = killExportProcess.waitForFinished();
+
+                if (!finishedKilling)
+                {
+                    killExportProcess.kill();
+                }
+
+                logStream << "Killing Project Export.";
```

Comment:
Shouldn't this be moved above to just before `killExportProcess.waitForFinished` ?

Reply:
No I don't think so. If I did that, it would prematurely dump the stdout as the kill process just got started. We have to wait for it to be finished so that stdout populates with the necessary log.